TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +56,77 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
             SyntaxProperty<SelectionModifier> selectionModifier = null;
             if (context.SQL_ALL() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All, ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
             }
             else if (context.SQL_DISTINCT() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct, ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
+
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var dotStarSelections = context.selections().selection().Select(selection => CreateDotStarSelection(selection)).Where(selection => selection != null).ToList();
+
+                return new SelectClause(selectionModifier, dotStarSelections);
+            }
+            return null;
+        }
+
+        private DotStarSelection CreateDotStarSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.dotStarSelection() != null)
+            {
+                var tableOrViewOrCorrelationName = context.dotStarSelection().tableOrViewOrCorrelationName();
+                if (tableOrViewOrCorrelationName.Name != null)
```

Comment:
I'm not so sure about this `if`. Event if Name is null, I think we should create a `DotStarSelection` instance (it would have a null `TableOrViewOrCorrelationName` property but that's ok) because it describes more precisely what's written. For example: `select tableName.*, .*` should lead to 2 `DotStarSelection` instance even though the second one is invalid.